### ISSUE TITLE
ShouldReturnUnit Rule

### ISSUE THREAD

**Issue by vanniktech** [arturbosch/detekt#1012]
Sometimes I'm still not used to being able to return `Unit` when I have a no-op function. It would be nice if such code would be triggered. Some examples of non compliant code:

```kotlin
override fun onUpgrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {
  // No-op for now.
}

override fun init() {
    
}
```

and their compliant versions:

```kotlin
override fun onUpgrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) = Unit

override fun init() = Unit
```

Making this an expression body with `Unit` makes it very clear what the intension is.

**Comment by arturbosch** [comment]
Hm I never thought about returning `Unit` when the default implementation should do nothing, interesting!

**Comment by marianosimone** [comment]
FWIW.. I'm still on the verge about this one. Some people on my team prefer:

```
fun foo() {
    // No-op
}
```

Which is definitely more explicit

However, I sometimes do:

```
fun foo() = Unit // No-op
```

Which makes it almost impossible to re-write the function without doing something about that comment.

Specially for newcomers to Kotlin, returning `Unit` is not clear at all that you meant not to implement the method, but I TOTALLY see this being a `detekt` rule that teams could turn on once people are used to the language.

**Comment by schalkms** [comment]
Agreed. This change should be made in the `OptionalUnit` rule.

**Comment by schalkms** [comment]
closed with f0b4d2e

**Comment by vanniktech** [comment]
Thank you!
